relentless/variable.py

```diff
@@ -74,13 +86,79 @@ class Variable(abc.ABC):
     not inherit from this variable directly but should make use of the more
     flexible :class:`IndependentVariable` or :class:`DependentVariable` types.
 
+    Examples
+    --------
+    Create variables with specified value::
+
+        x = relentless.variable.IndependentVariable(value=2.0)
+
+    If the value is numerical, arithmetic operations are available::
+
+        >>> print(x+2.0)
+        4.0
+        >>> print(5.0-x)
+        1.0
+        >>> print(2.0*x)
+        2.0
+        >>> print(x/4.0)
+        0.5
+        >>> print(x^2.0)
+        0.25
+        >>> print(-x)
+        -0.25
+
     """
     @property
     @abc.abstractmethod
     def value(self):
         """float: Value of the variable."""
         pass
 
+    def __add__(self, val):
+        """Addition of two variables, or of a variable and a scalar."""
+        v = val if isinstance(val, Variable) else IndependentVariable(val)
+        return Sum(self, val)
+
+    def __radd__(self, val):
+        """Addition of a scalar and a variable."""
+        return Sum(IndependentVariable(val), self)
```

Comment:
Is the reason `val` is always cast to an `IndependentVariable` because it must be something other than a `Variable` if `__radd__` gets called? Just asking because I don't remember when this is actually invoked in python.

Reply:
Yes, `1+var` is interpreted as `var.__radd__(1)` . If both operands are `Variable`s then the usual `__add__` method can be used.

Reply:
OK, that's what I thought, thanks!